src/components/eip3643/admin/RegisterIdentity.tsx

```diff
@@ -28,19 +39,45 @@ export default function RegisterIdentity({
     error,
     isPending,
   } = useRegisterIdentity();
+  let unsub: WatchContractEventReturnType | null = null;
 
   useEffect(() => {
     tokenSelected &&
       readTokenIdentityRegistry({ args: [] }, tokenSelected.address).then(
-        (res) => setRegistry(res[0]),
+        (res) => {
+          setRegistry(res[0]);
+          setRegistryAgents([]);
+          unsub && unsub();
+          unsub = watchIdentityRegistryIdentityRegisteredEvent(
+            {
+              onLogs: (data) => {
+                setRegistryAgents((prev: any) => {
+                  return [
+                    ...prev,
+                    ...data
+                      .map((item: any) => item.args[0])
+                      .filter((item) => !prev.includes(item)),
+                  ];
+                });
+              },
+            },
+            res[0] as `0x${string}`,
+          );
+        },
       );
+    return () => {
+      unsub && unsub();
+    };
   }, [tokenSelected]);
 
   if (!tokenSelected) return null;
-
   return (
     <>
       <Button
+        isDisabled={
+          !currentIdentityAddress ||
+          registryAgents.includes(currentIdentityWallet)
```

Comment:
we should indicate for user why the button is disabled, maybe some help text with "address X is already registered" or so

Reply:
Agreed, will think on it. Before that buttons were just hidden.